action/candidate_transfer_ownership.go

```diff
@@ -0,0 +1,206 @@
+	newOwner, err := address.FromString(newOwnerStr)
+	if err != nil {
+		return nil, err
+	}
+	return &CandidateTransferOwnership{
+		AbstractAction: AbstractAction{
+			version:  version.ProtocolVersion,
+			nonce:    nonce,
+			gasLimit: gasLimit,
+			gasPrice: gasPrice,
+		},
+		newOwner: newOwner,
+		payload:  payload,
+	}, nil
+}
+
+// NewCandidateTransferOwnershipFromAction decode data to CandidateTransferOwnership
+func NewCandidateTransferOwnershipFromABIBinary(data []byte) (*CandidateTransferOwnership, error) {
+	var (
+		paramsMap = map[string]any{}
+		ok        bool
+		err       error
+		cr        CandidateTransferOwnership
+	)
+	// sanity check
+	if len(data) <= 4 || !bytes.Equal(_candidateTransferOwnershipMethod.ID, data[:4]) {
+		return nil, errDecodeFailure
+	}
+	if err = _candidateTransferOwnershipMethod.Inputs.UnpackIntoMap(paramsMap, data[4:]); err != nil {
+		return nil, err
+	}
+	if cr.newOwner, err = ethAddrToNativeAddr(paramsMap["newOwner"]); err != nil {
+		return nil, err
+	}
+	if cr.payload, ok = paramsMap["payload"].([]byte); !ok {
+		return nil, errDecodeFailure
+	}
+	return &cr, nil
+}
+
+// Payload returns the payload bytes
+func (act *CandidateTransferOwnership) Payload() []byte { return act.payload }
+
+// NewOwner returns the new owner address
+func (act *CandidateTransferOwnership) NewOwner() address.Address { return act.newOwner }
+
+// IntrinsicGas returns the intrinsic gas of a CandidateTransferOwnership
+func (act *CandidateTransferOwnership) IntrinsicGas() (uint64, error) {
+	payloadSize := uint64(len(act.Payload()))
+	return CalculateIntrinsicGas(CandidateTransferOwnershipBaseIntrinsicGas, CandidateTransferOwnershipPayloadGas, payloadSize)
+}
+
+// Cost returns the total cost of a CandidateTransferOwnership
+func (act *CandidateTransferOwnership) Cost() (*big.Int, error) {
+	intrinsicGas, _ := act.IntrinsicGas()
+
+	fee := big.NewInt(0).Mul(act.GasPrice(), big.NewInt(0).SetUint64(intrinsicGas))
+	return fee, nil
+}
+
+// Serialize returns a raw byte stream of the CandidateTransferOwnership struct
+func (act *CandidateTransferOwnership) Serialize() []byte {
+	return byteutil.Must(proto.Marshal(act.Proto()))
+}
+
+// Proto converts to protobuf CandidateTransferOwnership Action
+func (act *CandidateTransferOwnership) Proto() *iotextypes.CandidateTransferOwnership {
+	ac := iotextypes.CandidateTransferOwnership{
+		NewOwnerAddress: act.newOwner.String(),
+	}
+
+	if len(act.payload) > 0 {
+		ac.Payload = make([]byte, len(act.payload))
+		copy(ac.Payload, act.payload)
+	}
+	return &ac
+}
+
+// LoadProto loads the CandidateTransferOwnership Action from protobuf
+func (act *CandidateTransferOwnership) LoadProto(pbAct *iotextypes.CandidateTransferOwnership) error {
+	if pbAct == nil {
+		return ErrNilProto
+	}
+	newOwner, err := address.FromString(pbAct.GetNewOwnerAddress())
+	if err != nil {
+		return err
+	}
+	act.newOwner = newOwner
+	act.payload = nil
+	if len(pbAct.GetPayload()) > 0 {
```

Comment:
`if payload := pbAct.GetPayload(); len(payload) > 0`
otherwise it is called 3 times